### PR TITLE
Feat access control

### PR DESCRIPTION
**Pull request by yang1666204**
## Summary

1. Page access permission control.
2. Support re-acquiring data after adding a new account/role.

### CODE REVIEW

File: ui/src/pages/Access/Accounts.tsx [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
+interface AccountsProps {
+  allAccounts: AcAccount[] | undefined;
+  refreshAccounts: () => void;
+}
+
+export default function Accounts({
+  allAccounts,
+  refreshAccounts,
+}: AccountsProps) {
+  const access = useAccess();
```
Comment (powerfooI) [inline ui/src/pages/Access/Accounts.tsx:21]: What would `useAccess()` return?

File: ui/src/access.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { AcAccount, AcPolicy } from './api/generated';
+import { initialAccess } from './utils/helper';
+
+export type InitialStateType = {
+  accountInfo: AcAccount;
+  policies: AcPolicy[];
+};
+
+export default function (initialState: InitialStateType) {
```
Comment (powerfooI) [inline ui/src/access.ts:9]: Added the return type to avoid implicit `any`

File: ui/src/pages/Layouts/BasicLayout/index.tsx [hunk 3]
```diff
@@ -55,11 +58,13 @@ const BasicLayout: React.FC = () => {
       }),
       link: '/tenant',
       icon: <IconFont type="tenant" />,
+      accessible: true,
```
Comment (powerfooI) [inline ui/src/pages/Layouts/BasicLayout/index.tsx:61]: This should be `access.obclusterread`

File: ui/src/pages/Layouts/BasicLayout/index.tsx [hunk 4]
```diff
@@ -91,16 +98,20 @@ const BasicLayout: React.FC = () => {
     </Menu>
   );
 
+  useEffect(() => {
+    history.push(menus.find((item) => item.accessible).link);
```
Comment (powerfooI) [inline ui/src/pages/Layouts/BasicLayout/index.tsx:102]: Why not use `history.replace()`? Users could go back if `push` is used here.

File: ui/src/pages/Layouts/DetailLayout/index.tsx [hunk 5]
```diff
@@ -84,6 +36,61 @@ const DetailLayout: React.FC<DetailLayoutProps> = ({
...
+      key: 'cluster',
+      link: '/cluster',
+      icon: <IconFont type="cluster" />,
+      accessible: access.obclusterread,
+    },
+    {
+      title: intl.formatMessage({
+        id: 'Dashboard.Cluster.Detail.Tenant',
+        defaultMessage: '租户',
+      }),
```
Comment (powerfooI) [inline ui/src/pages/Layouts/DetailLayout/index.tsx:63]: Add `accessible: access.obclusterread` here

File: ui/src/pages/Overview/index.tsx [hunk 6]
```diff
@@ -17,11 +19,13 @@ const OverviewPage: React.FC = () => {
       }}
     >
       <Row justify="start" gutter={[16, 16]}>
-        <OverviewStatus />
-        <Col span={24}>
-          <EventsTable />
-        </Col>
-        <NodesTable />
+        {access.systemread && <OverviewStatus />}
```
Comment (powerfooI) [inline ui/src/pages/Overview/index.tsx:22]: Merge the same three conditions here.

File: ui/src/utils/helper.ts [hunk 7]
```diff
@@ -116,3 +120,29 @@ export const reportPollData = async () => {
 export function floorToTwoDecimalPlaces(num: number) {
   return Math.floor(num * 100) / 100;
 }
+
+export function initialAccess(
```
Comment (powerfooI) [inline ui/src/utils/helper.ts:124]: Maybe the verb format `initializeAccess` is better

**Review by powerfooI — CHANGES_REQUESTED** [review]

File: ui/src/pages/Access/Accounts.tsx (hunk 1 above)
Comment (yang1666204) [reply ui/src/pages/Access/Accounts.tsx:21]: The object returned in access.ts. https://umijs.org/docs/max/access#useaccess

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**